### DISCUSSION TITLE
Can a single Separator handle concurrent separations?

### DISCUSSION THREAD

**Discussion by empz** [nomadkaraoke/python-audio-separator#101] [Q&A] [answered]
I have a web server that instantiates a single Separator and loads a model. When doing 1 separation at a time, everything works fine. As soon as I try to separate a second audio while the first is still being worked on I get weird behaviours, like getting None as the output files.
I have yet to dig into the code, but first I wanted to ask you if you know for sure it can't handle concurrent jobs.

Thanks!

**Comment by beveradb** [comment, marked as answer]
Hey @empz - in short, probably not!

I'm afraid I've never even considered trying to run two jobs in parallel, as typically the separation process consumes all resources on whatever system I'm running it on already!
I do actually run multiple jobs in parallel on Runpod, but those are in totally separate containers.

If you get it working somehow, please raise a PR! ☺️
